openvdb_houdini/houdini/SOP_OpenVDB_Advect_Points.cc

```diff
@@ -1102,18 +1107,26 @@ VDB_NODE_OR_CACHE(VDB_COMPILABLE_SOP, SOP_OpenVDB_Advect_Points)::evalAdvectionP
     UT_String ptGroupStr;
     evalString(ptGroupStr, "group", 0, now);
 
-    parms.mPointGroup = parsePointGroups(ptGroupStr, GroupCreator(gdp));
-    parms.mGroup = matchGroup(*parms.mPointGeo, evalStdString("group", now));
+    if (advectVdbPoints)
+    {
+	parms.mPointGroup = 0;
+	parms.mGroup = matchGroup(*parms.mPointGeo, ptGroupStr);
 
-    const std::string groups = evalStdString("vdbpointsgroups", now);
+	if (!parms.mPointGroup && ptGroupStr.length() > 0) {
+	    addWarning(SOP_MESSAGE, "Point group not found");
+	    return false;
+	}
+    }
+    else
+    {
+	parms.mPointGroup = parsePointGroups(ptGroupStr, GroupCreator(gdp));
+	parms.mGroup = 0;
 
-    // Get and parse the vdb points groups
-    openvdb::points::AttributeSet::Descriptor::parseNames(
-        parms.mIncludeGroups, parms.mExcludeGroups, evalStdString("vdbpointsgroups", now));
+	const std::string groups = evalStdString("vdbpointsgroups", now);
 
-    if (!parms.mPointGroup && ptGroupStr.length() > 0) {
-        addWarning(SOP_MESSAGE, "Point group not found");
-        return false;
+	// Get and parse the vdb points groups
+	openvdb::points::AttributeSet::Descriptor::parseNames(
+	    parms.mIncludeGroups, parms.mExcludeGroups, evalStdString("vdbpointsgroups", now));
```

Comment:
parms.mIncludeGroups and parms.mExcludeGroups are required to have been set in the other branch (when VDB Points are being advected)

Reply:
Yes, this does look backwards.   I'll have to make some new tests to ensure this code is exercised & I'll resubmit when I've verified one way or another.